File: Prolog/prolog1/gcd.pl
%Gabrielle Curcio
%CSC 435-01
%9/24/20
%Prolog 1: Problem 1
%This program finds the Greatest Common Divisor
%of 2 positive integers A and B
%
%Solution: Euclidean Algorithm
%Assume B > A
%If A = 0 and B is positive, then B is the GCD. Stop here.
%If A is positive, then R equals B mod A,where R is the remainder
%Repeat the above step with R as the new A and old A as the new B.

gcd(0,B,B) :- B>0.
gcd(A,B,D) :- A>0, R is B mod A, gcd(R,A,D).

File: Scala/scala1/gcd.scala
//Gabrielle Curcio
//CSC 435-01
//10/8/20
//Scala 1 - gcd
//This program determines the greatest common divisor
//of two integers entered by the user
import scala.io.StdIn.readInt

object Main {
  //this function determines the gcd using Euclid's algorithm
  def gcd(a:Int, b:Int): Int ={
    //Solve for the remainder using b mod a
    //If the remainder is 0, then b is the gcd
    //otherwise, recursively solve the gcd with 
    //a as the new b, and the remainder as the new a 
    if (a == 0) b
    else gcd(b%a,a)
  }

  def main(args: Array[String]): Unit = {
    //read two integers from command-line
    println("Please enter the 1st number:")
    val a = readInt()
    println("Please enter the 2nd number:")
    val b = readInt()
    //calculate the gcd
    val result = gcd(a,b)
    println("The greatest common divisor is: "+ result)
  }
}
